hudi-common/src/main/java/org/apache/hudi/common/schema/HoodieSchemaCompatibility.java

```diff
@@ -191,4 +215,42 @@ public static boolean areSchemasProjectionEquivalent(HoodieSchema schema1, Hoodi
     }
     return AvroSchemaUtils.areSchemasProjectionEquivalent(schema1.toAvroSchema(), schema2.toAvroSchema());
   }
+
+  /**
+   * Identifies the writer field that corresponds to the specified reader field.
+   * This function is adapted from AvroSchemaCompatibility#lookupWriterField
+   *
+   * <p>
+   * Matching includes reader name aliases.
+   * </p>
+   *
+   * @param writerSchema Schema of the record where to look for the writer field.
+   * @param readerField  Reader field to identify the corresponding writer field
+   *                     of.
+   * @return the writer field, if any does correspond, or None.
+   */
+  public static HoodieSchemaField lookupWriterField(final HoodieSchema writerSchema, final HoodieSchemaField readerField) {
+    assert (writerSchema.getType() == HoodieSchemaType.RECORD);
+    final List<HoodieSchemaField> writerFields = new ArrayList<>();
```

Comment:
It looks like the list is expected to have 0 or 1 elements so an Option may fit this usecase better, what do you think?